vowpalwabbit/cb_algs.cc

```diff
@@ -49,9 +49,14 @@ void predict_or_learn(cb& data, single_learner& base, example& ec)
 {
   CB::label ld = ec.l.cb;
   cb_to_cs& c = data.cbcs;
-  c.known_cost = get_observed_cost(ld);
-  if (c.known_cost != nullptr && (c.known_cost->action < 1 || c.known_cost->action > c.num_actions))
-    std::cerr << "invalid action: " << c.known_cost->action << std::endl;
+  auto optional_cost = get_observed_cost_cb(ld);
```

Comment:
it looks like we could skip the check here and just do `c.known_cost = optional_cost.second` since it will hold the default `CB::cb_class` if the cost is not known.

Reply:
I intentionally did not do this as if this were a true optional then the `.second` value would not be accessible

Reply:
True true